hts-sys/build.rs

```diff
@@ -43,114 +89,115 @@ fn main() {
+    let mut config_lines = vec![
+        "/* Default config.h generated by build.rs */",
+        "#define HAVE_DRAND48 1",
+    ];
+
     let use_bzip2 = env::var("CARGO_FEATURE_BZIP2").is_ok();
-    if !use_bzip2 {
-        let bzip2_patterns = vec!["s/ -lbz2//", "/#define HAVE_LIBBZ2/d"];
-        sed_htslib_makefile(&out, &bzip2_patterns, "bzip2");
-    } else if let Ok(inc) = env::var("DEP_BZIP2_ROOT")
+    if use_bzip2 {
+         if let Ok(inc) = env::var("DEP_BZIP2_ROOT")
         .map(PathBuf::from)
         .map(|path| path.join("include"))
-    {
-        cfg.include(inc);
+        {
+            cfg.include(inc);
+            config_lines.push("#define HAVE_LIBBZ2 1");
+        }
+    }
+
+    let use_libdeflater = env::var("CARGO_FEATURE_LIBDEFLATER").is_ok();
+    if use_libdeflater {
+        if let Ok(inc) = env::var("DEP_LIBDEFLATE_INCLUDE").map(PathBuf::from) {
+            cfg.include(inc);
+            config_lines.push("#define HAVE_LIBDEFLATE 1");
+        } else {
+            panic!("no DEP_LIBDEFLATE_INCLUDE");
+        }
     }
 
     let use_lzma = env::var("CARGO_FEATURE_LZMA").is_ok();
-    if !use_lzma && want_static {
-        let lzma_patterns = vec!["s/ -llzma//", "/#define HAVE_LIBLZMA/d"];
-        sed_htslib_makefile(&out, &lzma_patterns, "lzma");
-    } else if let Ok(inc) = env::var("DEP_LZMA_INCLUDE").map(PathBuf::from) {
-        cfg.include(inc);
+    if use_lzma {
+        if let Ok(inc) = env::var("DEP_LZMA_INCLUDE").map(PathBuf::from) {
+            cfg.include(inc);
+            config_lines.push("#define HAVE_LIBBZ2 1");
+            config_lines.push("#ifndef __APPLE__");
+            config_lines.push("#define HAVE_LZMA_H 1");
+            config_lines.push("#endif");
+        }
     }
 
     let use_curl = env::var("CARGO_FEATURE_CURL").is_ok();
-    if !use_curl {
-        let curl_patterns = vec!["s/ -lcurl//", "/#define HAVE_LIBCURL/d"];
-        sed_htslib_makefile(&out, &curl_patterns, "curl");
-    } else if let Ok(inc) = env::var("DEP_CURL_INCLUDE").map(PathBuf::from) {
-        cfg.include(inc);
+    if use_curl {
+        if let Ok(inc) = env::var("DEP_CURL_INCLUDE").map(PathBuf::from) {
+            cfg.include(inc);
+            config_lines.push("#define HAVE_LIBCURL 1");
+            cfg.file("htslib/hfile_libcurl.c");
+            println!("cargo:rerun-if-changed=htslib/hfile_libcurl.c");
+        }
     }
 
-    let tool = cfg.get_compiler();
-    let (cc_path, cflags_env) = (tool.path(), tool.cflags_env());
-    let cc_cflags = cflags_env.to_string_lossy().replace("-O0", "");
-    let host = env::var("HOST").unwrap_or_default();
-    // autoreconf & ./configure (with no args) steps are necessary to include the htslib plugins (hfile_s3.o, hfile_s3_writer.o, etc...)
-    if  // cleanup first
-        // TODO: Have top level "cargo clean" do this instead of in here, see:
-        // https://github.com/rust-lang/cargo/issues/572#issuecomment-632456478
```

Comment:
No need/wish to have `cargo clean`? I did affect my builds when re-compiling rust-htslib :-S

Reply:
The cargo bug is closed & it doesn't seem like it's going to be addressed.  Since we're moving the source into `OUT_DIR` and using `cc::Build.compile()` the object files and the copy of the source will actually get deleted by `cargo clean`, so I think we're good.